e2e/Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM golang:1.19 as builder
+
+WORKDIR /go/src/github.com/Azure/aks-app-routing-operator
+ADD ../. .
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go test -c -o e2e/e2e.test --count=1 --tags=e2e -c ./e2e
```

Comment:
not sure if you've tried but if we don't need CGO_ENABLED=0 we should remove it. It slows done build times a lot

Reply:
Can I ask what part didn't work when removing them? Curious if its the multi-stage using `scratch`?

Reply:
resolving after our discussion